### PR TITLE
[RFR] Create a REST v2 for getCollectionAction from PageController

### PR DESCRIPTION
**Pull request by ndufreche**
This PR is Work In Progress and depend of other updates, in particular in backbee/BbCoreJs.

Add new query param "root" and add the possibility ti use getCollectionAction as a real getCollection.

### CODE REVIEW

**Comment by mickaelandrieu** [comment]
your actual suite tests should pass or this "means" you have changed actual behavior, we can't allow BC atm

**Comment by ndufreche** [comment]
Yes I have change the actual behavior, because with the actual behavior is impossible to get a clean collection.

The other possibility is create an other route + action but I think soil the Rest Api to stay "BC" is not a better practice.

This is clearly a bug fix for me.

I have update unit test then you can see the new param an the more clean behavior.

**Comment by mickaelandrieu** [comment]
@ndufreche I understand your point but we can't break a REST API: it's far too late, the only viable option is to create another api call with a correct behavior and deprecate the actual for api v2.

Also this does not make sense to update the test in order to valid a new behavior : this way you can valid your tests all the time, test suite is a contract on features, you can change code but never change tests (except if your tests don't test anything).

For me, it's :-1:

**Comment by ndufreche** [comment]
I have update rest version to 2.

File: Rest/Controller/PageController.php [hunk 1]
```diff
@@ -151,6 +151,8 @@ public function getCollectionAction(Request $request, $start, $count, Page $pare
             throw new BadRequestHttpException(
                 'To get page collection by content, you must provide `content_uid` and `content_type` as query parameters.'
             );
+        } elseif ($request->attributes->get('version') == 2) {
```
Comment (ndufreche) [inline Rest/Controller/PageController.php]: @eric-chau there is a better solution to implement an version of the REST api ?

**Comment by ndufreche** [comment]
@mickaelandrieu I have done lot of upgrade and add micro and temporally version managing REST version.
Version 1 still totally compatible with the day 1 getCollectionAction response (test have been restored).
Version 2 take lot of improvements and is now a better getCollectionAction implementation (new test have been added).

**Comment by mickaelandrieu** [comment]
There is no perfect solution, I agree to merge this pull request from my side :+1: 

Would you mind to rebase your commits ? /c @eric-chau are you ok ?

**Comment by ndufreche** [comment]
Yes I'm agree, is not the best implementation, in particular the version management. But is the best I can do currently.

**Comment by mickaelandrieu** [comment]
@ndufreche can you squash your commits ? => time to merge : - )

**Comment by ndufreche** [comment]
Ok I do it soon.

**Comment by ndufreche** [comment]
Before squash and merge I will add a last parameter to the V2 
 - `has_children` to return only Sections.

**Comment by mickaelandrieu** [comment]
thank you @ndufreche

**Comment by ndufreche** [comment]
backbee/backbee-standard#326 solve in part with the attribute has_children

**Pull request merged**